Add interfaces/wake-lock.idl

autofoolip opened the pull request:
This PR was automatically created by a bot.

Before merging, please check that any tests that depend on the updated IDL files still work.

If additional changes are needed, please manually create another PR based on this one.

See the [README](https://github.com/web-platform-tests/wpt/blob/master/interfaces/README.md) for how the IDL files in this directory are used.

<hr>

Source: https://github.com/tidoust/reffy-reports/blob/d030e72/ed/idl/wake-lock.idl
Build: https://travis-ci.org/tidoust/reffy-reports/builds/687558831

foolip commented:
I've filed https://github.com/tidoust/reffy/issues/303 to get this sorted in Reffy.

autofoolip committed:
Add interfaces/wake-lock.idl

Source: https://github.com/tidoust/reffy-reports/blob/d030e72/ed/idl/wake-lock.idl
Build: https://travis-ci.org/tidoust/reffy-reports/builds/687558831

stephenmcgruer commented:
@foolip so what is the next step here? Tidoust appears to claim this is fixed, but autofoolip still has this PR and https://github.com/web-platform-tests/wpt/pull/23107 open. Are we expecting them to be automatically closed, or do we need to do something here?

foolip commented:
A problem was fixed, namely wake-lock.idl is being updated again. However, reffy-reports will keep using the TR name, and that's still wake-lock. I would recommend one of:
- letting it be wake-lock.idl in reffy-reports and wpt until https://www.w3.org/TR/screen-wake-lock/ exists
- renaming this specifically in https://github.com/tidoust/reffy-reports/blob/master/wpt-sync/sync.js (leading to reffy-reports and wpt mismatch)
- land https://github.com/tidoust/reffy/pull/304 (leading to reffy-reports's [tr/idl/wake-lock.idl](https://github.com/tidoust/reffy-reports/blob/master/tr/idl/wake-lock.idl) going stale)
- a shortname override mechanism in reffy, initially used only for this

I would go for the first :)
